### PR TITLE
Fix incorrect wrap in sentence with linkReference

### PR DESCRIPTION
**Pull request by seiyab**
## Description
Fix https://github.com/prettier/prettier/issues/16351

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] ~(If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).~
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-markdown/printer-markdown.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ function genericPrint(path, options, print) {
         parts.push([parts.pop(), doc]);
         continue;
       }
-      parts.push(doc);
+      parts.push(doc, "");
```
Comment (seiyab) [inline src/language-markdown/printer-markdown.js:69]: `parts` should always satisfy following:
- its length is odd
- element in even index should be line

In this path, `doc` is line. To meet the former rule, we need additional element (`""`).

**Comment by fisker** [comment]
I don't think we build the doc correctly, [check this link](https://deploy-preview-16546--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAVAFgSwM4AIBmEANgCZwBOu6AhngNoBqAEgMoC6uM1ADnAZkTh4YEXGHJxq8XAHNM+PIUox0fOgHVMJGXBi4W6CAHcwtOGwB0uVKPIBXKJ1UBbADS5MUbFyJFczFgJyCGdcZ2pPXAAKFT5Y5z5qKBJcbHE4BGxDPUhncOS8aglcKAg9AE9dXAlBMxIASgsQVxAIbhhMaGxkUCLgowAFIszkEGoiI2py7paAI3JqMABrXRZqBIAZTzhkfHHsODmF5dXuRc8ZZBh7Q5AD50wrm5a4AA9eckwE2HHUCigipghLt9rdsBdBABFOxlHZIPZEA4tABW2FeLAhcGhsJBiNuAEcYfABsFuN0ULQALRQDJkEjNEDXCJEC4AYRC4VG4yIDPBUBkggAgjBrphZnZiRQtjTcUiQOgYM4iOosPBsGcwHAWJlMB0AG668qjMDYGYgPV2OAASWSCBgLHEmHaguSLBg5UEstu3GCB3UC24ox9Qgoep2LU8B3IMBJ1BknPhoJaZ3IUdG4XISxIxigDJ9nhgmhIKmQAA4AAwtCSEzASWPx6helpcWZFktIABMLTsB1Q1Fm5IRcrgzlmcBIdI2SRkdjjcAAYhByOERRcuRKICAAL5boA) (Click <kbd>show doc</kbd>)


```json
[
  fill([
    [
      "This",
      " ",
      "folder",
      " ",
      "has",
      " ",
      "[",
      "VHS",   <- Problem
      "]",
      " ",
      "tape",
      " ",
      "files",
      " ",
      "to",
      " ",
      "create",
      " ",
      "gifs",
      " ",
      "for",
      " ",
      "the",
      " ",
      "[",
      "Widget",
      " ",
      "Showcase",
      "]",
      ".",
      " ",
      "To",
      " ",
      "run",
      " ",
      "them,",
      " ",
      "install",
      " ",
      "VHS",
      " ",
      "from",
      " ",
      "main",
      " ",
      "(the",
      " ",
      "theme",
      " ",
      "and",
      " ",
      "screenshot",
      " ",
      "commands",
      " ",
      "are",
      " ",
      "not",
      " ",
      "yet",
      " ",
      "released).",
    ],
  ]),
  hardline,
]
```

**Comment by seiyab** [comment]
Thanks, I'll take a look later.

**Comment by fisker** [comment]
Sorry, please dismiss, I was using `--prose-wrap=preserve`...

**Comment by seiyab** [comment]
@fisker
Isn't it array in fill that doesn't need to satisfy rule of fill?
(Overrunning should be valid since proseWrap is preserve)

File: src/language-markdown/printer-markdown.js (hunk 1 above)
Comment (fisker) [reply src/language-markdown/printer-markdown.js:69]: Can you add a comment here? and L52?

**Comment by fisker** [comment]
Thanks @seiyab , looks good.

Would you like to add a stricter assertion to `fill` (followup PR)? We planed to do this before, but there are violations at that time.

File: changelog_unreleased/markdown/16546.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+<!-- Prettier stable -->
+This folder has [VHS] tape files to create gifs for the [Widget Showcase]. To run
+them, install VHS from main (the theme and screenshot commands are not yet released).
+
+<!-- Prettier main -->
+This folder has [VHS] tape files to create gifs for the [Widget Showcase]. To
+run them, install VHS from main (the theme and screenshot commands are not yet
+released).
+```
```
Comment (fisker) [inline changelog_unreleased/markdown/16546.md:20]:
````suggestion
<!-- prettier-ignore -->
```md
<!-- Input (--prose-wrap=always) -->
This folder has [VHS] tape files to create gifs for the [Widget Showcase]. To run them, install VHS from main (the theme and screenshot commands are not yet released).

<!-- Prettier stable -->
This folder has [VHS] tape files to create gifs for the [Widget Showcase]. To run
them, install VHS from main (the theme and screenshot commands are not yet released).

<!-- Prettier main -->
This folder has [VHS] tape files to create gifs for the [Widget Showcase]. To
run them, install VHS from main (the theme and screenshot commands are not yet
released).
```
````

**Comment by seiyab** [comment]
@fisker 
> Would you like to add a stricter assertion to fill (followup PR)?

Thank you for suggestion, maybe I will 🙋

**Review by fisker — APPROVED** [review]

**Review by kachkaev — APPROVED** [review]

**Comment by kachkaev** [comment]
Thank you for this contribution @seiyab!

**Comment by seiyab** [comment]
@fisker 
I found that we still have violations especially in CSS.
[Link for playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEATAlgNwATADpTbYC2AhgE4Dm6UAtADZwBmMS2tAAoywLIXVQA3ASIxypKAGcmEcsTZiJk+qXgBNABScITJpLgwAlMKgBfAiAA0ICAAcY6aJOSgK5CAHcAChQTOUpPQepACeztYARuJgANYGAMqkxHAAMjRwyEyB+pHRcTDxtqRgNJTIYgCucNb6xOjl5FXWcAAetnDk6MmwgQAqHVAU6HD+WfQ5IJKljACKFRDwmdnVIABWki3x03BzCxlIYxMAjvPwXu62-iCkkrRQcHCoj1Ygiuj0pQDCEMRkyNf0egvKZQSiMACCMDE6AiFTOHTS9yW4xWAAsYMR6AB1VHoeCSIpgODxPx4rB4kL-MCScIgTBVACSUCesHiYE69nBzPiMBCjGRE1s7n0WPEtn+gWCYReNH05Bg51IlD+B2W1iK5DlVJpLyFNBgWPQqBgqOQAA4AAzWchwE7oG2K5WkAUrGCkCKG42mpAAJmsFX0vXdozVIDgxAijyeqBSEkoFSVcAAYrIyFDShK4RAQKZTEA) (It has `fill(["-", ["@", "leftMargin"]])`)

I'll investigate it. I may open an issue if I find that we have many violations, so that I can manage and share problems / progress / findings.

**Pull request merged**